src/NLog/Targets/FileTarget.cs

```diff
@@ -1096,8 +1100,13 @@ protected virtual string GetFormattedMessage(LogEventInfo logEvent)
         /// <returns>Array of bytes that are ready to be written.</returns>
         protected virtual byte[] GetBytesToWrite(LogEventInfo logEvent)
         {
-            string renderedText = this.GetFormattedMessage(logEvent) + this.NewLineChars;
-            return this.TransformBytes(this.Encoding.GetBytes(renderedText));
+            string text = this.GetFormattedMessage(logEvent);
+            int textBytesCount = this.Encoding.GetByteCount(text);
+            int newLineBytesCount = this.Encoding.GetByteCount(this.NewLineChars);
+            byte[] bytes = new byte[textBytesCount + newLineBytesCount];
+            this.Encoding.GetBytes(text, 0, text.Length, bytes, 0);
+            this.Encoding.GetBytes(this.NewLineChars, 0, this.NewLineChars.Length, bytes, textBytesCount);
```

Comment:
@nazim9214 

As the NewLineChars & Encoding almost never changes while running, I was thinking to "cache" this one (`static` variable). I noticed before while performance testing that setting a fixed bytes for the newline improves the performance ~10-20%.  (but with a limited test)

Your thoughts please :)